src/main/scala/strawman/collection/BuildFrom.scala

```diff
@@ -0,0 +1,88 @@
+package strawman.collection
+
+import strawman.collection.mutable.Builder
+
+import scala.{Any, Ordering}
+
+/**
+  * Auxiliary type allowing us to define which collection type to build according to the type of an existing collection.
+  *
+  * @tparam From Existing collection type that will drives the collection type to build
+  * @tparam A Element type
+  */
+trait BuildFrom[-From, -A] {
+
+  /** Collection type to build */
+  type To
+
+  def newBuilder(): Builder[A, To]
+
+}
+
+trait BuildFromLowPriority {
+
+  /**
+    * Extracts the binary type constructor of `From` and applies it to `A` and `B`
+    * to build the `To` type.
+    */
+  implicit def binaryTC[CC[_, _], A, B](implicit
+    cb: CanBuild[(A, B), CC[A, B]]
+  ): BuildFrom.Aux[CC[_, _], (A, B), CC[A, B]] =
+    new BuildFrom[CC[_, _], (A, B)] {
+      type To = CC[A, B]
+      def newBuilder(): Builder[(A, B), CC[A, B]] = cb.newBuilder()
+    }
+
+}
+
+object BuildFrom extends BuildFromLowPriority {
+
+  type Aux[From, A, To0] = BuildFrom[From, A] { type To = To0 }
```

Comment:
I agree. I would try to use parameters everywhere because that's how we started. Also, if we bring back CanBuildFrom in a way, it's best not to do arbitrary changes to it.